compiler/src/iree/compiler/Pipelines/Options.cpp

```diff
@@ -122,14 +135,23 @@ void PreprocessingOptions::bindOptions(OptionsBinder &binder) {
 void GlobalOptimizationOptions::bindOptions(OptionsBinder &binder) {
   static llvm::cl::OptionCategory category(
       "IREE options for controlling global optimizations.");
+  auto init_at_opt = binder.optimizationLevel(
+      GlobalPipelineOptions::kGlobalOptFlag,
```

Comment:
Given that there can only be a single top level flag this is a footgun we need to keep hidden. Your binder can store the flag that was called with topLevelOpt, or we can hardcode it with a placeholder if you need it out of order (any optimizationLevel added before a topLevelOpt was defined gets added to the dummy option info, and then when topLevelOpt is called if it's already registered that gets swapped out). Having to reference flags by name in distributed code (plugins/etc) has design issues that are best to avoid.